jbpm-flow/src/main/java/org/jbpm/process/instance/context/variable/VariableScopeInstance.java

```diff
@@ -162,5 +168,41 @@ public void setContextInstanceContainer(ContextInstanceContainer contextInstance
     		this.variableInstanceIdPrefix = ((CompositeContextNodeInstance) contextInstanceContainer).getUniqueId();
     	}
 	}
+    
+    public void enforceRequiredVariables() {
+        VariableScope variableScope = getVariableScope();
+        for (Variable variable : variableScope.getVariables()) {
+            String name = variable.getName();
+            if (variableScope.isRequired(name)) {  
+                // check case file if it is prefixed
+                if (name.startsWith(VariableScope.CASE_FILE_PREFIX)) {
+                    if (!findCaseData(variable.getName())) {
+                        throw new VariableViolationException(getProcessInstance().getId(), name, "Case file item '" + name + "' is required but not set");
+                        
+                    }
+                    // otherwise check variables                    
+                } else if (!variables.containsKey(name)) {
+                    throw new VariableViolationException(getProcessInstance().getId(), variable.getName(), "Variable '" + variable.getName() + "' is required but not set");
```

Comment:
Same here, `variable.getName() `is already at `name`, and also processInstanceId can be added into the errorMessage

Reply:
fixed